### ISSUE TITLE
dotnet ef migrations - error when migrating from IdentityServer4 v4.1 to Duende IdentityServer v6.0

### ISSUE THREAD

**Issue by gunnars04** [DuendeSoftware/Support#996]
**Which version of Duende IdentityServer are you using?**
 IdentityServer4 v4.1
**Which version of .NET are you using?**
.net 6.0
**Describe the bug**

I‘m following locally this documentation:
https://docs.duendesoftware.com/identityserver/v6/upgrades/is4_v4_to_dis_v6/

But in step 5 when running either of these commands:
dotnet ef migrations add UpdateToDuende_v6_0 -c PersistedGrantDbContext -o Data/Migrations/IdentityServer/PersistedGrantDb
dotnet ef migrations add UpdateToDuende_v6_0 -c ConfigurationDbContext -o Data/Migrations/IdentityServer/ConfigurationDb

First I got this error:
"An unexpected return type was encountered while accessing the Microsoft.Extensions.Hosting services. Method 'CreateHostBuilder(string[])' should return an object of type 'IHostBuilder'. Continuing without the application service provider."

Then I fixed my program.cs. After that I get this error:
"An error occurred while accessing the Microsoft.Extensions.Hosting services. Continuing without the application service provider. Error: Value cannot be null. (Parameter 'certificate')"

When google‘in the problem, this article: https://ghanshyamshukla.medium.com/migration-identityserver4-to-duende-identityserver6-5bf0c6e58c3f
pointed to these sql migration files:
https://github.com/DuendeSoftware/IdentityServer/blob/main/migrations/IdentityServerDb/Migrations/ConfigurationDb.sql
https://github.com/DuendeSoftware/IdentityServer/blob/main/migrations/IdentityServerDb/Migrations/PersistedGrantDb.sql
but looking at the commit history, these latest files seem to be for a newer Identityserver then Identityserver 6.0

Do you know what might be causing the error? Or can I perhaps instead use sql migration scripts directly (skip  the „dotnet ef migrations add …“) and if so, which script version on github should I use for upgrading to IdentityServer v6.0 for Microsoft SQL Server?

Thanks in advance.

**To Reproduce**

Steps to reproduce the behavior.

**Expected behavior**

**Log output/exception with stacktrace**

```
data
```

**Additional context**

**Comment by AndersAbel** [comment]
It looks like something is not correctly set up in your environment for Entity Framework to be able to create the migrations. When creating a migration, Entity Framework starts the application briefly to get some configuration values. It looks like something prevents that startup from running correctly. To troubleshoot, it is often easiest to first make sure that IdentityServer can start normally. If you try to run your IdentityServer setup in your local development environment, does it start?

The migration scripts in those GitHub links are initial migration scripts that creates an empty database. They cannot be used to migrate an existing database.

**Comment by gunnars04** [comment]
@AndersAbel 
But would running these work for Identityserver 4.1 to 6 migration?
https://github.com/DuendeSoftware/IdentityServer/blob/main/migrations/IdentityServerDb/Migrations/ConfigurationDb.sql
https://github.com/DuendeSoftware/IdentityServer/blob/main/migrations/IdentityServerDb/Migrations/PersistedGrantDb.sql

**Comment by AndersAbel** [comment]
No. Not if you want to keep any data in your IdSrv4 database. Those scripts start from an empty database and create empty tables.

**Comment by gunnars04** [comment]
@AndersAbel :

When running:
dotnet ef migrations add UpdateToDuende_v6_0 -c PersistedGrantDbContext -o Data/Migrations/IdentityServer/PersistedGrantDb
https://docs.duendesoftware.com/identityserver/v6/upgrades/is4_v4_to_dis_v6/

My program.cs was like this and then I got this error:
`        public static IWebHostBuilder BuildWebHost(string[] args) =>
            WebHost.CreateDefaultBuilder(args)
                .UseStartup<Startup>()
                .ConfigureLogging(logging =>
                {
                    logging.ClearProviders();
                    logging.SetMinimumLevel(Microsoft.Extensions.Logging.LogLevel.Trace);
                })
                .UseNLog();  // NLog: Setup NLog for Dependency injection`
                
 `
Build started...
Build succeeded.
An unexpected return type was encountered while accessing the Microsoft.Extensions.Hosting services. Method 'CreateHostBuilder(string[])' should return an object of type 'IHostBuilder'. Continuing without the application service provider.
Unable to create an object of type 'PersistedGrantDbContext'. For the different patterns supported at design time, see https://go.microsoft.com/fwlink/?linkid=851728`

I changed it to:
`        public static IHostBuilder CreateHostBuilder(string[] args) =>
            Host.CreateDefaultBuilder(args)
                    .ConfigureWebHostDefaults(webBuilder =>
                    {
                        webBuilder.UseStartup<Startup>()
                        .ConfigureLogging(logging =>
                        {
                            logging.ClearProviders();
                            logging.SetMinimumLevel(Microsoft.Extensions.Logging.LogLevel.Trace);
                        })
                        .UseNLog();  // NLog: Setup NLog for Dependency injection
                    });`
                    
Was that correct?

**Comment by gunnars04** [comment]
Works now, thanks! 👍

**Comment by gunnars04** [comment]
I just needed to add:
options.ConfigureDbContext = builder => builder.UseSqlServer(identityServerConnectionString, b => b.MigrationsAssembly("NameOfMyIdentityServerProject"));
